### DISCUSSION TITLE
How to specify a release version?

### DISCUSSION THREAD

**Discussion by thadguidry** [jbangdev/jbang#1544] [Q&A] [answered]
Hi we have a new project at https://github.com/homihq/homi-cli
We want to make it easy for users to download and try new versions.
We are versed in publishing releases using GitHub Actions and also to Maven Central, etc.

Using jbang, how can a user specify that they want a particular version like 1.0 or 2.0 if we start publishing releases on GitHub?
Or must we publish to Maven Central and users specify a coordinate somehow?

For example, I'm a Windows user, with a fresh install or virtual environment and want to tell my users to just do something like this:

`jbang homi@homi-cli/1.1.0`
or
`jbang com.homi:homi-cli:1.1.0`

Ya know, like NPM easy?

I see the full syntax is `<alias>@<user/org>(/repository)(/branch)(~path)`.
So perhaps we can use branches/tags on GitHub? sorta like my first example line above?
Thanks in advance for any pointers or developer best practices for publishing and then downloading a particular version of an application.

**Comment by quintesse** [comment, marked as answer]
You could definitely use `jbang com.homi:homi-cli:1.1.0` if that was published on Maven Central. That's perhaps the easiest from the PoV of Jbang usage but getting things published on Maven Central isn't _that_ trivial.

You can use Jbang's catalog and aliases for something like `jbang homi@homi-cli` but it doesn't have a versioning system. You could perhaps make aliases with the version in the name, eg: `jbang homi-1-1-0@homi-cli`.

Another option could indeed be to use GitHub tags and use `jbang homi@homi-cli/jbang-catalog-or-some-other-name/1.1.0`, but the name is getting a bit long perhaps and it might not be so nice if you have different projects in that catalog all following different versionings.

A final option is to use the Maven syntax, like the first option, but use JitPack, like this: `jbang com.github.homihq:homi-cli:1.1.0`.

Edit: fixed mistake in command: `jbang homi@homi-cli-1-1-0` is not correct, should be: `jbang homi-1-1-0@homi-cli`.
